src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -189,14 +182,6 @@ public function normalize($data, string $format = null, array $context = [])
      */
     public function denormalize($data, string $type, string $format = null, array $context = [])
     {
-        if (isset(self::SCALAR_TYPES[$type])) {
-            if (!('is_'.$type)($data)) {
-                throw new NotNormalizableValueException(sprintf('Data expected to be of type "%s" ("%s" given).', $type, get_debug_type($data)));
-            }
-
-            return $data;
-        }
-
```

Comment:
This is a breaking change because code that relied on that logic now has to configure an additional normalizer.

Reply:
You're right, when I started working on this I thought this change was a lot more recent and I forgot to consider the impact.
I'll try to find a way to revert that so that the PR won't break anything but maybe we should deprecate this use case ?

Reply:
I put the code back and added a condition to only use it the the `ScalarDenormalizer` is not registered.
I also added a deprecation and tried to add the necessary information in the different files as mentioned in the conventions.